### DISCUSSION TITLE
I don't like the new Releases UI.

### DISCUSSION THREAD

**Discussion by sblantipodi** [community/community#6138] [Other Feature Feedback, Questions, & Ideas] [answered]
As title.
New UI is simply less useful.

I tend to add a complete changelog into the release.
New UI shows only two rows of the changelog and users needs to click read more on every releases to see what's new.

I simply find it less useful than before.

**Comment by TomGalla11** [comment]
I have the same opinion as you! The new UI isn‘t that good. If you format the changelog like code then no lines are shown. I know many developers who do it like this. I also like it when I see the whole changelog and not just a few lines! The tag button is also missing, the tags are very useful to get all releases on „one“ site. The new design is basically the old UI - (minus) buttons, text,...

**Comment by ann0see** [comment]
Yes. Totally agree. IMHO a good release page consists of
1. A prominent download button (or multiple) which can be enabled/disabled by developers, pointed to assets, automatically detects OS,... Currently they are just too much hidden. Which user wants to read the whole changelog, click on an invisible button, search through multiple assets and then find a downloadable binary?
2. A changelog (which is below the download button)

**Comment by Triky313** [comment]
Images are also no longer displayed. Only when you have clicked Read more.

In addition, a new page loads when I click on Read more, which is not a nice solution.

**Comment by HeCorr** [comment]
Agreed, it shows a lot less information, plus there's a lot of wasted space between the left and right sections.

**Comment by MylesBorins** [comment]
We are planning a number of improvements that should land in the next week or so to drastically improve the experience

* Showing the full notes for the latest release
* Making "Read More" show the full release notes rather than forward to another page
* An attribute to show full releases rather than abridged
* Showing the full notes for all releases in search results
* Highlighting search terms in search results

**Reply by panva** [reply]
@MylesBorins 

> An attribute to show full releases rather than abridged

Is this an attribute the UI is going to "remember"?

**Reply by MylesBorins** [reply]
At his point it will not, but we do expand by default on search results. you can add the attribute `&expanded=true` to any release pages to see the full contents

**Reply by panva** [reply]
@MylesBorins without the option remembered I find my personal use of this UI as impeded compared to the current one.

**Reply by jsunsawyer** [reply]
@MylesBorins I'd prefer that full notes were shown for all releases, not just the latest.

**Reply by mertant** [reply]
> Making "Read More" show the full release notes rather than forward to another page

This is how it currently seems to work for me, and it is not satisfactory. I agree with this statement:

> I'd prefer that full notes were shown for all releases, not just the latest.

I use the Releases page to browse through releases/drafts and scan through the change notes of each. The old UI supported my workflow; the current version requires me to click "Read more" on every release individually. So this new UI has actively made things worse for me.

A marked improvement would be to have **an "Expand All" button/setting on the Releases page** (whose value is stored). This would support my workflow without requiring repetitive "Read more" clicks, while still leaving the current style as the default for those people who do like the "collapsed" release notes.

(As a side note - in terms of vertical space, the release's footer box is rather massive. The "Contributors", "Assets", and "Reactions" elements altogether (even when fully collapsed) make the footer box actually taller than the 3-4 lines of text preview. If I was concerned about vertical space, that's where I would start.)

**Reply by sblantipodi** [reply]
@MylesBorins I still don't like it. What's the point of those changes?
What do you "improved" with those changes?
The old release page was good enough and very functional. 
Now you changed something, but you improved nothing.

**Comment by MylesBorins** [comment]
Hey all. We made a few more improvements and would love it if you took a look. In particular we ran some numbers on the average size of releases and found that 90% of releases have less than 547 characters in the body. We've made it so that we only shorten releases that are more than 750 characters, which should result in the vast majority of GitHub releases remaining un affected by the changes.

At the same time we have now shipped:

* We do not abridge releases who have a body of less than 750 characters (over 90% of all GitHub releases)
* Making "Read More" show the full release notes rather than forward to another page
* An attribute to show full releases rather than abridged
* Showing the full notes for all releases in search results

and are currently working on:

* Highlighting search terms in search results

We are separately trying to improve the algorithm to determine the latest release to make it more accurate / expected, in the mean time we have paused showing the full releases for latest releases.

**Reply by jsunsawyer** [reply]
@MylesBorins Perhaps it's not clear, but the inconsistency in things being expanded versus collapsed is what's causing the issue for many of us.

Personally, I don't want to use search, so any improvements to search are lost on me. I just want to do a quick browser find in the releases page. Unfortunately, it feels like we're being pushed to use search.

That said, introducing a new inconsistency to how releases handle "Read More" only seems to create an edge case where users who use browser find may end up not finding results in the page because they're hidden, even though the user may assume all notes are expanded.

@mertant outlined the functionality that should have been implemented here:

> A marked improvement would be to have an "Expand All" button/setting on the Releases page (whose value is stored). This would support my workflow without requiring repetitive "Read more" clicks, while still leaving the current style as the default for those people who do like the "collapsed" release notes.

This functionality may be less useful based on your updated logic, and potentially confusing since _most_ releases are now expanded by default.

**Reply by MylesBorins** [reply]
@jsunsawyer I can definitely appreciate where you are coming from. We are trying to find the balance here between verbosity and scalability and it is a tough balance. When projects have incredibly long release notes it is extremely difficult to scroll through the page and see all the releases, and while a quick search of the frame can help you pin-point specific content on the page it does make it harder for other folks who are trying to scan high level details of releases.

We did a variety of user-research and there was quite a bit of feedback that we received letting us know that the prior page was difficult to navigate for a number of folks trying to solve different use cases. As we saw much of the use of the full content being in finding details in releases and finding details across pages of releases was even harder (don't get me started on the old pagination) we opted to invest in search both for pagination and discoverability of particular content.

I recognize that this is a change in workflow for folks and that is frustrating, but I am not personally convinced that we should return to showing the full content by default, nor do I think that putting a "expand all" button at the top of the page is the right solution to the problem.

One thing that we are exploring is adding [keyboard-shortcuts](https://docs.github.com/en/get-started/using-github/keyboard-shortcuts) for our power users who want to speed up their specific use of releases. Here are a few of the ideas we have been tossing around for potential shortcuts:

* expand all
* focus search
* prev / next page
* prev / next release (on release landing page

I have a distinct feeling that aside from "expand all" or "just expanding everything by default" you are not going to find these solution satisfying, but i would be interested in hearing if you might be open to alternative workflows

**Reply by jsunsawyer** [reply]
@MylesBorins

> When projects have incredibly long release notes it is extremely difficult to scroll through the page and see all the releases

This seems like an exaggeration. Sure, the occasional release might be long, but scrolling is far from difficult, and you're introducing a new issue by making it difficult to utilize native browser search to quickly search for different terms in the page's content. Not only are you creating a new issue, but you're changing something that a ton of people are used to. That has to account for something, right?

Would it not make more sense to simply allow users to collapse notes on a release with lots of notes (or any release), similar to comment threads in a forum? This would allow users to scan the full content of recent releases (allowing cmd + f to be usable in a consistent manor – no need to ensure that a page's content is visible), but allow users who are searching for high level release notes to quickly bypass notes for a release that wasn't relevant to them, and move on to the next.

I'm not quite sure why you don't just add a way for users to manage and save the expanded state (?expanded=true), as that would resolve this issue for several of us. See https://github.com/github/feedback/discussions/6515 for more context.

If the following settings existed, I'm curious what users would pick:
- Show full release notes
- Show partial release notes
- Hide release notes

I'm sorry, but I pick "Show full release notes" every time, leaving it up to the author to determine how long their release notes are.

Additionally, when using search instead of cmd + f, you lose some important context, as you only see releases with matching search terms. With cmd + f, you can find your matching term in a release's notes, then look at the surrounding releases. I'm not saying that search should account for this, but rather pointing out the additional utility of cmd + f.

> We did a variety of user-research and there was quite a bit of feedback that we received letting us know that the prior page was difficult to navigate for a number of folks trying to solve different use cases.

I trust that you've researched this, and that you're using feedback from users, but is this not what feature previews and feedback discussions are for? Your last line in your response to me is extremely disheartening.

**Reply by mertant** [reply]
> I recognize that this is a change in workflow for folks and that is frustrating, but I am not personally convinced that we should return to showing the full content by default, 

I'm fine with "my way" not being the default, just something that is opt-in. (As long as it only has to be opted in once, not every time the page is navigated to). Whether the opting in is something that's done on the Releases page, or in Settings, doesn't matter much, because it should be something that I only ever do once (on that particular browser/user account, at least).

> nor do I think that putting a "expand all" button at the top of the page is the right solution to the problem.

Why is that? It seems to me that the button (or a toggle, etc.) would be a functional solution, at least, even if it is a basic one. If the objection to that is on aesthetic grounds (an extra button would make the UI too busy/messy/etc.), then, as mentioned above, a different implementation like hiding it in the Settings page would do just fine. I don't have any particular need or desire to press a button - I have a need to be able to scroll through full release notes. 

> One thing that we are exploring is adding keyboard-shortcuts for our power users who want to speed up their specific use of releases. Here are a few of the ideas we have been tossing around for potential shortcuts:
> 
> expand all
> focus search
> prev / next page
> prev / next release (on release landing page

A keyboard shortcut could achieve the same thing, although they are less discoverable than buttons and settings flags. 

Still, ideally, the value of the flag would be remembered, because pressing a key just one time is better than pressing it every time you navigate onto a page. (Although GitHub does seem to tend towards not remembering the position of your toggles. For example, the "Hide whitespace" option when viewing a PR has to be applied again manually every time.)

On the topic of searching, I only have this to say: you can't search-by-keyword for something if you don't know that it exists in the first place. Only browsing can do that.

**Reply by MylesBorins** [reply]
hey @jsunsawyer I've been reflecting a bunch over the last day on our conversation, in particular:

> Your last line in your response to me is extremely disheartening.

I never want to discourage folks or give them the impression that their feedback isn't welcome, for that I apologize.

As per [my most recent update](https://github.com/github/feedback/discussions/6138#discussioncomment-1555668) we are going to be rolling back the truncated release list experience based on accessibility review. Thank you again for providing feedback.

**Reply by jsunsawyer** [reply]
Much appreciated, @MylesBorins.

**Comment by MylesBorins** [comment, marked as answer]
Hey All,

Yet another update!

We did a review with an accessibility expert internally at GitHub this week and found that the current truncated experience does not play nicely with screen readers after much consideration we are going to be removing truncated releases from the default experience in the next week.

We still think the truncated experience is something we want a pursue, but the time needed to make a screen reader compatible experience is too great at this moment. We are going to instead pivot our energy to cleaning up the few remaining bugs we are aware of and hopefully getting highlighted terms in search results.

I believe with the removal of truncated releases we have now closed the loop on the majority of specific feedback items that were brought up in this thread. I'm going to mark this as the answer and will update again when the changes have landed.

**Reply by ocornut** [reply]
Since a few months GitHub went back to truncated Changelog.
IMHO it is become particularly bad because:
- the truncation is BARELY visible (consist in a "..." at the end of a set of potentially large paragraph)
- there's no way to uncollapse/untruncate from the truncation point.

The only way to uncollapse/untruncate is to SCROLL BACK UP, click Version Number and go to a different page.

I consider this a severe regression in making Changelog / Release Notes widely available.
Previously it was possible to have a page with all our Changelog readily available in one location, now it is impossible.

I have opened a Ticket about it but thought it would make sense to write this down here too.

**Reply by Die4Ever** [reply]
why do collapsed sections count against the character limit? I use a lot of <details></details> in my release notes, but it's still easy to end up truncated no matter what I do and how condensed I actually make it

**Comment by sblantipodi** [comment]
Thank you for listening to us. I love GitHub!!!
